consensus/istanbul/validator/multi_staking_test.go

```diff
@@ -170,7 +170,9 @@ func TestCalcTotalAmount(t *testing.T) {
 		stakingAmounts := testCase.stakingAmounts
 		totalAmount := calcTotalAmount(testCase.weightedValidators, testCase.stakingInfo, stakingAmounts)
 
-		assert.Equal(t, testCase.expectedGini, testCase.stakingInfo.Gini)
+		// Since calcTotalAmount does not update stakingInfo.Gini, we cannot assert expectedGini.
+		// However asserting expectedStakingAmounts proves that calcTotalAmount() used correct Gini.
+		// assert.Equal(t, testCase.expectedGini, testCase.stakingInfo.Gini)
```

Comment:
I didn't get why this testcase is failed. Shouldn't this pass?

Reply:
Before this PR, this test expects `calcTotalAmount()` to fill in the `stakingInfo.Gini`.
After this PR, `calcTotalAmount()` should not change the `stakingInfo.Gini`.

The tests actually fail because `testCase.expectedGini = 0.27` but `testCase.stakingInfo.Gini = -1`.

Reply:
@jeongkyun-oh To make it more clear, I will modify `calcTotalAmount` to return the `gini` (`giniUsed` might be more appropriate naming) as well, and then compare with `testCase.expectedGini`.

```go
totalAmount, giniUsed := calcTotalAmount(...)
assert.Equal(t, testCase.expectedGini, giniUsed) // calcTotalAmount computes gini on its own among validators
assert.Equal(t, params. DefaultGiniCoefficient, testCase.stakingInfo.Gini) // stakingInfo.Gini left untouched
```

Reply:
@blukat29 Wouldn't it `calcTotalAmount` change the normal behaviour to calculate weight of validators?

Reply:
@jeongkyun-oh Does changing `calcTotalAmount()` affect the behavior of `calcWeight()`? No because `calcWeight()` does not depend on `stakingInfo`.

As long as `calcTotalAmount()` returns correct `stakingAmounts` and `totalStaking`, `calcWeight()` should keep working correctly.

Reply:
Oh, never mind. Applying your suggestion is good enough. Thanks.